### PR TITLE
Moved all helper methods to a different file

### PR DESCRIPTION
**Pull request by snigdha-px**
**What this PR does / why we need it**:
https://portworx.atlassian.net/browse/PA-560
**Which issue(s) this PR fixes** (optional)
Closes #
1)Moved all helper function in a seperate file
2)Added return function for all methods

**Special notes for your reviewer**:

Jenkins link : https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test/164/consoleFull
https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/PX-Backup/job/system-tests/job/px-backup-system-test/112/consoleFull
http://aetos.pwx.purestorage.com/resultSet/testSetID/99076

### CODE REVIEW

File: tests/backup/backup_helper_test.go [hunk 1]
```diff
@@ -0,0 +1,1205 @@
+package tests
```
Comment (lsrinivas-pure) [inline tests/backup/backup_helper_test.go:1]: do not mark as test file, name it as common

File: tests/backup/backup_helper_test.go [hunk 2]
```diff
@@ -0,0 +1,1205 @@
...
+func getProviderClusterConfigPath(provider string, kubeconfigs []string) (string, error) {
+	log.Infof("Get kubeconfigPath from list %v and provider %s",
+		kubeconfigs, provider)
+	for _, kubeconfigPath := range kubeconfigs {
+		if strings.Contains(provider, kubeconfigPath) {
+			fullPath := path.Join(KubeconfigDirectory, kubeconfigPath)
+			return fullPath, nil
+		}
+	}
+	return "nil", fmt.Errorf("kubeconfigPath not found for provider %s in kubeconfigPath list %v",
```
Comment (lsrinivas-pure) [inline tests/backup/backup_helper_test.go:62]: return "" string

File: tests/backup/backup_helper_test.go [hunk 3]
```diff
@@ -0,0 +1,1205 @@
...
+			Uid:  postRuleUid,
+		},
+	}
+	_, err := backupDriver.CreateBackup(ctx, bkpCreateRequest)
+	if err != nil {
+		return err
+	}
+	backupSuccessCheck := func() (interface{}, bool, error) {
+		bkpUid, err = backupDriver.GetBackupUID(ctx, backupName, orgID)
+		log.FailOnError(err, "Failed while trying to get backup UID for - %s", backupName)
```
Comment (lsrinivas-pure) [inline tests/backup/backup_helper_test.go:104]: return err

File: tests/backup/backup_helper_test.go [hunk 4]
```diff
@@ -0,0 +1,1205 @@
...
+		backupInspectRequest := &api.BackupInspectRequest{
+			Name:  backupName,
+			Uid:   bkpUid,
+			OrgId: orgID,
+		}
+		resp, err := backupDriver.InspectBackup(ctx, backupInspectRequest)
+		log.FailOnError(err, "Inspecting the backup taken with request:\n%v", backupInspectRequest)
```
Comment (lsrinivas-pure) [inline tests/backup/backup_helper_test.go:111]: retrun err

File: tests/backup/backup_helper_test.go [hunk 5]
```diff
@@ -0,0 +1,1205 @@
...
+		if actual != expected {
+			return "", true, fmt.Errorf("backup status for [%s] expected was [%s] but got [%s]", backupName, expected, actual)
+		}
+		return "", false, nil
+	}
+
+	task.DoRetryWithTimeout(backupSuccessCheck, maxWaitPeriodForBackupCompletionInMinutes*time.Minute, 30*time.Second)
+
+	bkpUid, err = backupDriver.GetBackupUID(ctx, backupName, orgID)
+	log.FailOnError(err, "Failed while trying to get backup UID for - %s", backupName)
```
Comment (lsrinivas-pure) [inline tests/backup/backup_helper_test.go:123]: return err

File: tests/backup/backup_helper_test.go [hunk 6]
```diff
@@ -0,0 +1,1205 @@
...
+		},
+		ResourceTypes: resourceType,
+	}
+	_, err := backupDriver.CreateBackup(ctx, bkpCreateRequest)
+	if err != nil {
+		return err
+	}
+	backupSuccessCheck := func() (interface{}, bool, error) {
+		bkpUid, err = backupDriver.GetBackupUID(ctx, backupName, orgID)
+		log.FailOnError(err, "Failed while trying to get backup UID for - %s", backupName)
```
Comment (lsrinivas-pure) [inline tests/backup/backup_helper_test.go:193]: return err

File: tests/backup/backup_helper_test.go [hunk 7]
```diff
@@ -0,0 +1,1205 @@
...
+		backupInspectRequest := &api.BackupInspectRequest{
+			Name:  backupName,
+			Uid:   bkpUid,
+			OrgId: orgID,
+		}
+		resp, err := backupDriver.InspectBackup(ctx, backupInspectRequest)
+		log.FailOnError(err, "Inspecting the backup taken with request:\n%v", backupInspectRequest)
```
Comment (lsrinivas-pure) [inline tests/backup/backup_helper_test.go:200]: return err

File: tests/backup/backup_helper_test.go [hunk 8]
```diff
@@ -0,0 +1,1205 @@
...
+			Name: preRuleName,
+			Uid:  preRuleUid,
+		},
+		PostExecRuleRef: &api.ObjectRef{
+			Name: postRuleName,
+			Uid:  postRuleUid,
+		},
+	}
+	_, err := backupDriver.CreateBackup(ctx, bkpCreateRequest)
+	log.FailOnError(err, "Failed to take backup with request -\n%v", bkpCreateRequest)
```
Comment (lsrinivas-pure) [inline tests/backup/backup_helper_test.go:341]: return err

File: tests/backup/backup_helper_test.go [hunk 9]
```diff
@@ -0,0 +1,1205 @@
...
+	groupIDs := make([]string, 0)
+	userIDs := make([]string, 0)
+
+	bkpUid, err := backupDriver.GetBackupUID(ctx, backupName, orgID)
+	log.FailOnError(err, "Failed to get backup UID for [%s]", backupName)
+	log.Infof("Backup UID for %s - %s", backupName, bkpUid)
+
+	for _, groupName := range groupNames {
+		groupID, err := backup.FetchIDOfGroup(groupName)
+		log.FailOnError(err, "Error fetching group ID")
```
Comment (lsrinivas-pure) [inline tests/backup/backup_helper_test.go:357]: return err

File: tests/backup/backup_helper_test.go [hunk 10]
```diff
@@ -0,0 +1,1205 @@
...
+// addUsersOrGroups - provide false if the mentioned users/groups needs to be deleted or removed
+func ClusterUpdateBackupShare(clusterName string, groupNames []string, userNames []string, accessLevel BackupAccess, addUsersOrGroups bool, ctx context.Context) error {
+	backupDriver := Inst().Backup
+	groupIDs := make([]string, 0)
+	userIDs := make([]string, 0)
+	_, clusterUID := backupDriver.RegisterBackupCluster(orgID, SourceClusterName, "")
+
+	for _, groupName := range groupNames {
+		groupID, err := backup.FetchIDOfGroup(groupName)
+		log.FailOnError(err, "Error fetching group ID")
```
Comment (lsrinivas-pure) [inline tests/backup/backup_helper_test.go:413]: return err

File: tests/backup/backup_helper_test.go [hunk 11]
```diff
@@ -0,0 +1,1205 @@
...
+	backupNames := make([]string, 0)
+	backupDriver := Inst().Backup
+	ctx, err := backup.GetNonAdminCtx(username, password)
+	if err != nil {
+		return nil, nil
+	}
+
+	bkpEnumerateReq := &api.BackupEnumerateRequest{
+		OrgId: orgID}
+	curBackups, err := backupDriver.EnumerateBackup(ctx, bkpEnumerateReq)
```
Comment (lsrinivas-pure) [inline tests/backup/backup_helper_test.go:483]: handle the err here

File: tests/backup/backup_helper_test.go [hunk 12]
```diff
@@ -0,0 +1,1205 @@
...
+	return backupNames, err
+}
+
+func GetNodesForBackup(backupName string, bkpNamespace string,
+	orgID string, clusterName string, triggerOpts *driverApi.TriggerOptions) ([]node.Node, error) {
+
+	var nodes []node.Node
+	backupDriver := Inst().Backup
+	backupUID, err := getBackupUID(backupName, orgID)
+	dash.VerifyFatal(err, nil, fmt.Sprintf("Getting backuip UID for backup %s", backupName))
```
Comment (lsrinivas-pure) [inline tests/backup/backup_helper_test.go:496]: return err

File: tests/backup/backup_helper_test.go [hunk 13]
```diff
@@ -0,0 +1,1205 @@
...
+		if err != nil {
+			return err
+		}
+		ctx, err := backup.GetAdminCtxFromSecret()
+		if err != nil {
+			return err
+		}
+		_, err = DeleteBackup(backupName, backupUID, OrgID, ctx)
+		if err != nil {
+			return err
```
Comment (lsrinivas-pure) [inline tests/backup/backup_helper_test.go:662]: do not return err if you want to handle the failures using dash else return err at all the places, just to be consistent

File: tests/backup/backup_helper_test.go [hunk 14]
```diff
@@ -0,0 +1,1205 @@
...
+		case 8:
+			access = FullAccess
+			role = backup.InfrastructureOwner
+		default:
+			access = ViewOnlyAccess
+			role = backup.ApplicationOwner
+		}
+		ctxNonAdmin, err := backup.GetNonAdminCtx(users[i], "Password1")
+		log.FailOnError(err, "Fetching user ctx")
+		userRoleAccessContext := userRoleAccess{users[i], role, access, ctxNonAdmin}
```
Comment (lsrinivas-pure) [inline tests/backup/backup_helper_test.go:887]: return err

File: tests/backup/backup_helper_test.go [hunk 15]
```diff
@@ -0,0 +1,1205 @@
...
+	return restoreNames, nil
+}
+
+// TODO: There is no delete org API
+/*func DeleteOrganization(orgID string) {
+	Step(fmt.Sprintf("Delete organization [%s]", orgID), func() {
+		backupDriver := Inst().Backup
+		req := &api.Delete{
+			CreateMetadata: &api.CreateMetadata{
+				Name: orgID,
```
Comment (lsrinivas-pure) [inline tests/backup/backup_helper_test.go:1072]: remove the commented part of not required

File: tests/backup/backup_helper_test.go [hunk 16]
```diff
@@ -0,0 +1,1205 @@
...
+	}
+	response, err := backupDriver.EnumerateBackupLocation(ctx, backupLocationEnumerateRequest)
+	if err != nil {
+		return backupLocationMap, err
+	}
+	if len(response.BackupLocations) > 0 {
+		log.Infof("Found backup locations - ")
+		for _, backupLocation := range response.BackupLocations {
+			backupLocationMap[backupLocation.Uid] = backupLocation.Name
+			log.Infof(backupLocation.Name)
```
Comment (lsrinivas-pure) [inline tests/backup/backup_helper_test.go:1161]: add more to the log statement or remove it if it is added for debuging

File: tests/backup/backup_helper_test.go [hunk 17]
```diff
@@ -0,0 +1,1205 @@
...
+	}
+	response, err := backupDriver.EnumerateCloudCredential(ctx, cloudCredentialEnumerateRequest)
+	if err != nil {
+		return cloudCredentialMap, err
+	}
+	if len(response.CloudCredentials) > 0 {
+		log.Infof("Found cloud credentials - ")
+		for _, cloudCredential := range response.CloudCredentials {
+			cloudCredentialMap[cloudCredential.Uid] = cloudCredential.Name
+			log.Infof(cloudCredential.Name)
```
Comment (lsrinivas-pure) [inline tests/backup/backup_helper_test.go:1183]: same here

**Review by lsrinivas-pure — CHANGES_REQUESTED** [review]

File: tests/backup/backup_helper_test.go (hunk 2 above)
Comment (snigdha-px) [reply tests/backup/backup_helper_test.go:62]: done

File: tests/backup/backup_helper_test.go (hunk 3 above)
Comment (snigdha-px) [reply tests/backup/backup_helper_test.go:104]: done

File: tests/backup/backup_helper_test.go (hunk 4 above)
Comment (snigdha-px) [reply tests/backup/backup_helper_test.go:111]: done

File: tests/backup/backup_helper_test.go (hunk 5 above)
Comment (snigdha-px) [reply tests/backup/backup_helper_test.go:123]: done

File: tests/backup/backup_helper_test.go (hunk 6 above)
Comment (snigdha-px) [reply tests/backup/backup_helper_test.go:193]: done

File: tests/backup/backup_helper_test.go (hunk 7 above)
Comment (snigdha-px) [reply tests/backup/backup_helper_test.go:200]: done

File: tests/backup/backup_helper_test.go (hunk 8 above)
Comment (snigdha-px) [reply tests/backup/backup_helper_test.go:341]: done

File: tests/backup/backup_helper_test.go (hunk 9 above)
Comment (snigdha-px) [reply tests/backup/backup_helper_test.go:357]: done!

File: tests/backup/backup_helper_test.go (hunk 10 above)
Comment (snigdha-px) [reply tests/backup/backup_helper_test.go:413]: done!!

File: tests/backup/backup_helper_test.go (hunk 11 above)
Comment (snigdha-px) [reply tests/backup/backup_helper_test.go:483]: done

File: tests/backup/backup_helper_test.go (hunk 12 above)
Comment (snigdha-px) [reply tests/backup/backup_helper_test.go:496]: done!

File: tests/backup/backup_helper_test.go (hunk 13 above)
Comment (snigdha-px) [reply tests/backup/backup_helper_test.go:662]: done

File: tests/backup/backup_helper_test.go (hunk 14 above)
Comment (snigdha-px) [reply tests/backup/backup_helper_test.go:887]: done

File: tests/backup/backup_helper_test.go (hunk 16 above)
Comment (snigdha-px) [reply tests/backup/backup_helper_test.go:1161]: done

File: tests/backup/backup_helper_test.go (hunk 17 above)
Comment (snigdha-px) [reply tests/backup/backup_helper_test.go:1183]: done

File: tests/backup/backup_helper_test.go (hunk 15 above)
Comment (snigdha-px) [reply tests/backup/backup_helper_test.go:1072]: done

File: tests/backup/backup_test.go [hunk 18]
```diff
@@ -5165,7 +5174,8 @@ var _ = Describe("{RestoreEncryptedAndNonEncryptedBackups}", func() {
 		ctx, err = backup.GetAdminCtxFromSecret()
 		log.FailOnError(err, "Fetching px-central-admin ctx")
 		for _, backupName := range backupNames {
-			backupUID := getBackupUID(backupName, orgID)
+			backupUID, err := getBackupUID(backupName, orgID)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Getting backuip UID for backup %s", backupName))
```
Comment (lsrinivas-pure) [inline tests/backup/backup_test.go:5178]: backup spell check

File: tests/backup/backup_test.go [hunk 19]
```diff
@@ -5448,8 +5459,9 @@ var _ = Describe("{ReplicaChangeWhileRestore}", func() {
 			err := Inst().S.Destroy(contexts[i], opts)
 			dash.VerifySafely(err, nil, fmt.Sprintf("Verify destroying app %s", taskName))
 		}
-		backupUID := getBackupUID(backupName, orgID)
-		_, err := DeleteBackup(backupName, backupUID, orgID, ctx)
+		backupUID, err := getBackupUID(backupName, orgID)
+		dash.VerifyFatal(err, nil, fmt.Sprintf("Getting backuip UID for backup %s", backupName))
```
Comment (lsrinivas-pure) [inline tests/backup/backup_test.go:5463]: backup spell check

File: tests/backup/backup_test.go [hunk 20]
```diff
@@ -5300,8 +5310,9 @@ var _ = Describe("{CustomResourceBackupAndRestore}", func() {
 			dash.VerifyFatal(err, nil, fmt.Sprintf("Deleting Restore %s", restore))
 		}
 		for _, backupName := range backupNames {
-			backupUID := getBackupUID(backupName, orgID)
-			_, err := DeleteBackup(backupName, backupUID, orgID, ctx)
+			backupUID, err := getBackupUID(backupName, orgID)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Getting backuip UID for backup %s", backupName))
```
Comment (lsrinivas-pure) [inline tests/backup/backup_test.go:5314]: backup spell check

File: tests/backup/backup_test.go (hunk 18 above)
Comment (snigdha-px) [reply tests/backup/backup_test.go:5178]: done

File: tests/backup/backup_test.go (hunk 19 above)
Comment (snigdha-px) [reply tests/backup/backup_test.go:5463]: done

File: tests/backup/backup_test.go (hunk 20 above)
Comment (snigdha-px) [reply tests/backup/backup_test.go:5314]: done

File: tests/backup/backup_helper_test.go (hunk 1 above)
Comment (snigdha-px) [reply tests/backup/backup_helper_test.go]: done

File: tests/backup/common.go [hunk 21]
```diff
@@ -0,0 +1,1306 @@
...
+		userName := fmt.Sprintf("testuser%v-%v", i, time.Now().Unix())
+		firstName := fmt.Sprintf("FirstName%v", i)
+		lastName := fmt.Sprintf("LastName%v", i)
+		email := fmt.Sprintf("%v@cnbu.com", userName)
+		wg.Add(1)
+		go func(userName, firstName, lastName, email string) {
+			err := backup.AddUser(userName, firstName, lastName, email, password)
+			Inst().Dash.VerifyFatal(err, nil, fmt.Sprintf("Creating user - %s", userName))
+			users = append(users, userName)
+			wg.Done()
```
Comment (lsrinivas-pure) [inline tests/backup/common.go:933]: make defer wg.Done and move to first line

Comment (lsrinivas-pure) [reply tests/backup/common.go:933]: also add defer ginkgorecover

File: tests/backup/common.go [hunk 22]
```diff
@@ -0,0 +1,1306 @@
...
+		return backupLocationMap, err
+	}
+	if len(response.BackupLocations) > 0 {
+		for _, backupLocation := range response.BackupLocations {
+			backupLocationMap[backupLocation.Uid] = backupLocation.Name
+		}
+	} else {
+		log.Info("No backup locations found")
+	}
+	log.Infof("The backup locations and their UID are %v", backupLocationMap)
```
Comment (lsrinivas-pure) [inline tests/backup/common.go:1267]: move this to inside if condition

File: tests/backup/common.go [hunk 23]
```diff
@@ -0,0 +1,1306 @@
...
+		return cloudCredentialMap, err
+	}
+	if len(response.CloudCredentials) > 0 {
+		for _, cloudCredential := range response.CloudCredentials {
+			cloudCredentialMap[cloudCredential.Uid] = cloudCredential.Name
+		}
+	} else {
+		log.Info("No cloud credentials found")
+	}
+	log.Infof("The backup locations and their UID are %v", cloudCredentialMap)
```
Comment (lsrinivas-pure) [inline tests/backup/common.go:1288]: same as above

File: tests/backup/backup_test.go [hunk 24]
```diff
@@ -6138,7 +6152,7 @@ var _ = Describe("{IssueMultipleRestoresWithNamespaceAndStorageClassMapping}", f
 		log.InfoD("Deleting the backup created")
 		backupDriver := Inst().Backup
 		backupUID, err := backupDriver.GetBackupUID(ctx, backupName, orgID)
-		log.FailOnError(err, "Error getting backup UID for %v", backupName)
+		dash.VerifySafely(err, nil, "Deleting the backup created")
```
[the remark this answers is not in the document]
Comment (snigdha-px) [reply tests/backup/backup_test.go:6155]: done

File: tests/backup/common.go (hunk 21 above)
Comment (snigdha-px) [reply tests/backup/common.go:933]: done

File: tests/backup/common.go (hunk 22 above)
Comment (snigdha-px) [reply tests/backup/common.go:1267]: done

File: tests/backup/common.go (hunk 23 above)
Comment (snigdha-px) [reply tests/backup/common.go:1288]: done

File: tests/backup/common.go [hunk 25]
```diff
@@ -0,0 +1,1308 @@
...
+		if restoreResponseStatus.GetStatus() == api.RestoreInfo_StatusInfo_PartialSuccess || restoreResponseStatus.GetStatus() == api.RestoreInfo_StatusInfo_Success {
+			log.Infof("Restore status - %s", restoreResponseStatus)
+			log.InfoD("Status of %s - [%s]",
+				restoreName, restoreResponseStatus.GetStatus())
+			return "", false, nil
+		}
+		return "", true, fmt.Errorf("expected status of %s - [%s] or [%s], but got [%s]",
+			restoreName, api.RestoreInfo_StatusInfo_PartialSuccess.String(), api.RestoreInfo_StatusInfo_Success, restoreResponseStatus.GetStatus())
+	}
+	task.DoRetryWithTimeout(restoreSuccessCheck, 10*time.Minute, 30*time.Second)
```
Comment (lsrinivas-pure) [inline tests/backup/common.go:754]: please have a look and handle the error

**Review by lsrinivas-pure — CHANGES_REQUESTED** [review]

File: tests/backup/common.go (hunk 25 above)
Comment (snigdha-px) [reply tests/backup/common.go]: done

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Comment by mkoppal-px** [comment]
Instead of naming it `common.go`, can we name it `backup_helper.go` ?
It may avoid getting confused with the `tests/common.go`

**Comment by mkoppal-px** [comment]
Remove all the unused constants, variables and functions from `tests/backup/common.go`

File: tests/backup/common.go [hunk 26]
```diff
@@ -0,0 +1,1318 @@
...
+	lastName                                  = "lastName"
+	password                                  = "Password1"
+)
+
+var (
+	// enable all the after suite actions
+	wantAllAfterSuiteActions bool = true
+	// selectively enable after suite actions by setting wantAllAfterSuiteActions to false and setting these to true
+	wantAfterSuiteSystemCheck     bool = false
+	wantAfterSuiteValidateCleanup bool = false
```
Comment (mkoppal-px) [inline tests/backup/common.go:61]: Remove the type here

Comment (snigdha-px) [reply tests/backup/common.go:61]: done

**Comment by snigdha-px** [comment]
> Remove all the unused constants, variables and functions from `tests/backup/common.go`

done

**Comment by snigdha-px** [comment]
> Instead of naming it `common.go`, can we name it `backup_helper.go` ? It may avoid getting confused with the `tests/common.go`

done

**Review by vpinisetti-px — COMMENTED** [review]
LGTM

**Review by sn-px — APPROVED** [review]
LGTM

File: tests/backup/backup_helper.go [hunk 27]
```diff
@@ -0,0 +1,1234 @@
...
+	}
+	return roleAccessUserBackupContext, nil
+}
+func ValidateSharedBackupWithUsers(user string, access BackupAccess, backupName string, restoreName string) {
+	ctx, err := backup.GetAdminCtxFromSecret()
+	Inst().Dash.VerifyFatal(err, nil, "Fetching px-central-admin ctx")
+	userCtx, err := backup.GetNonAdminCtx(user, "Password1")
+	Inst().Dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching %s user ctx", user))
+	log.InfoD("Registering Source and Destination clusters from user context")
+	CreateSourceAndDestClusters(orgID, "", "", userCtx)
```
Comment (mkoppal-px) [inline tests/backup/backup_helper.go:953]: Handle the error here

File: tests/backup/backup_helper.go [hunk 28]
```diff
@@ -0,0 +1,1234 @@
...
+		}
+		ctxNonAdmin, err := backup.GetNonAdminCtx(users[i], "Password1")
+		if err != nil {
+			return nil, err
+		}
+		userRoleAccessContext := userRoleAccess{users[i], role, access, ctxNonAdmin}
+		roleAccessUserBackupContext[userRoleAccessContext] = backupNames[i]
+		err = backup.AddRoleToUser(users[i], role, "Adding role to user")
+		if err != nil {
+			err = fmt.Errorf("Failed to add role %s to user %s with err %v", role, users[i], err)
```
Comment (mkoppal-px) [inline tests/backup/backup_helper.go:936]: Make all letter lower case in error message

File: tests/common.go [hunk 29]
```diff
@@ -1975,26 +1975,21 @@ func updateNamespace(in interface{}, namespaceMapping map[string]string) error {
...
-		//  fmt.Sprintf("Failed to delete cloud credential [%s] in org [%s]", name, orgID))
-		// TODO: validate CreateCloudCredentialResponse also
-	})
+	backupDriver := Inst().Backup
+	credDeleteRequest := &api.CloudCredentialDeleteRequest{
+		Name:  name,
+		OrgId: orgID,
+		Uid:   cloudCredUID,
+	}
+	//ctx, err := backup.GetPxCentralAdminCtx()
```
Comment (mkoppal-px) [inline tests/common.go:1986]: Remove this commented line

File: tests/backup/backup_helper.go (hunk 27 above)
Comment (snigdha-px) [reply tests/backup/backup_helper.go:953]: done

File: tests/backup/backup_helper.go (hunk 28 above)
Comment (snigdha-px) [reply tests/backup/backup_helper.go:936]: done

File: tests/common.go (hunk 29 above)
Comment (snigdha-px) [reply tests/common.go:1986]: done

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Pull request merged**